### PR TITLE
ISPN-11174 Fix RestMetricsResource test failure

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-11174

### CODE REVIEW

**Comment by tristantarrant** [comment]
testMicroprofileMetricsMetadata is still failing, but at least it is not attempting to create multiple servers anymore

File: server/runtime/src/test/java/org/infinispan/server/functional/RestMetricsResource.java [hunk 1]
```diff
@@ -35,10 +34,7 @@
 public class RestMetricsResource {
 
    @ClassRule
-   public static final InfinispanServerRule SERVERS =
-         InfinispanServerRuleBuilder.config("configuration/ClusteredServerTest.xml")
-                                    .numServers(2)
```
Comment (anistor) [inline server/runtime/src/test/java/org/infinispan/server/functional/RestMetricsResource.java]: My intention was to ensure we have at least 2 nodes running to ensure duplicate metric names in same registry, which then gets de-duplicated by the unique node tag.

Comment (tristantarrant) [reply server/runtime/src/test/java/org/infinispan/server/functional/RestMetricsResource.java]: But the ClusteredIT is already running two nodes, and RestMetricsResource is being run as part of its suite

Comment (anistor) [reply server/runtime/src/test/java/org/infinispan/server/functional/RestMetricsResource.java]: oh, did not realize that :)

**Comment by anistor** [comment]
Merged. Thanks @tristantarrant !

**Comment by anistor** [comment]
I've cherrypicked this to the 10.1.x port of the metrics PR also

**Pull request merged**